api/app/models/user_role.rb

```diff
@@ -0,0 +1,6 @@
+class UserRole < ApplicationRecord
+  self.primary_key = [ :user_id, :role_id ]
```

Comment:
As far as I remember, this setting is not necessary for Rails join tables and may cause issues.